docs/proposals/control-data-plane-split/README.md

```diff
@@ -0,0 +1,263 @@
+
+An example of what this looks like is defined [here](https://docs.nginx.com/nginx/admin-guide/load-balancer/dynamic-configuration-api/#configuring-persistence-of-dynamic-configuration).
+
+#### Secret duplication and synchronization
+
+There are multiple Secrets that an NGINX Plus user can and will be creating. These include:
+
+- JWT Secret for running NGINX Plus
+- Docker Registry Secret for pulling NGINX Plus
+- Client cert/key Secret for NIM connection
+- CA cert Secret for NIM connection
+
+With the new architecture, a user should initially create those Secrets in the nginx-gateway namespace. The control plane will then need to duplicate these Secrets to any namespace where it deploys an nginx instance. This is because the Secrets are mounted to the nginx deployment. The control plane should also update the Secrets if the original Secrets are ever updated, meaning it will have to now watch for Secret updates.
+
+This process must be documented so users are aware that their Secrets are being duplicated into other namespaces.
+
+### Encryption
+
+The agent and control plane communication channel will be encrypted. We will store the server certificate, key pair, and
+CA certificate in Kubernetes Secrets. The server Secret will live in the `nginx-gateway` namespace, and the agent Secret will live in the same namespace where the agent is deployed. The Secrets need to exist before the control plane and data planes are deployed.
+
+- `nginx-gateway-cert`: This Secret will contain the TLS certificate and private key that the control plane will use to
+  serve gRPC traffic.
+- `nginx-agent-cert`: This Secret will contain the CA bundle that validates the control plane’s certificate.
+
+The Secrets will be mounted to the control plane and agent containers, respectively. If desired, we can make the Secret
+names and mount path configurable via flags. For production, we will direct the user to provide their own certificates.
+For development and testing purposes, we will provide a self-signed default certificate. In order to be secure by
+default, NGF should generate the default certificates and keypair during installation using a Kubernetes Job.
+
+Using cert-manager may also be an easy option to reduce the burden of installing and rotating Secrets. A user would need to install this before NGF, and ensure they create agent Secrets before deploying their Gateway resource. We could also tie the NGF control plane directly into cert-manager so that our control plane could create the agent Secrets for the user when they create a Gateway resource, further reducing the burden on the user.
```

Comment:
The encryption solution is going to be interesting...we want to make the setup process as easy as we can for a user.

Should we fully manage the certificate lifecycle? Should a user not be involved in it at all since it's our communication channel, not theirs?

Reply:
Just below this I also added the SPIRE option.

Reply:
I guess we can just go with the approach listed earlier. We deploy with self-signed, and document a way for users to integrate cert-manager to use in production.

Updated the language here a bit.